ludwig/schema/model_types/base.py

```diff
@@ -88,12 +91,17 @@ def from_dict(config: ModelConfigDict) -> "ModelConfig":
         if isinstance(backend, str):
             config[BACKEND] = {"type": backend}
 
+        # JSON schema validation.
+        check_schema(config)
```

Comment:
May no longer be necessary to call this when loading the `ModelConfig` object.

Reply:
Marshmallow deserialization permits additional properties by default, while JSON schema validation doesn't when `additionalProperties=False`, which is the case for at least some of our schema (e.g `trainer`). I added a comment in the code to give a bit more context. 

That said, if we are able to set `additionalProperties=False` for all Ludwig schema, which seemed desirable based on an offline discussion with @ksbrar (cc: @connor-mccorm), then it might be possible to get to behavior parity marshmallow deserialization by using `schema.load(unknown='RAISE')`.